internal/mqttconf/will.go

```diff
@@ -0,0 +1,44 @@
+package mqttconf
+
+import (
+	"errors"
+
+	"github.com/Jeffail/benthos/v3/internal/docs"
+)
+
+// Will holds configuration for the last will message that the broker emits,
+// should benthos exit abnormally.
+type Will struct {
+	QoS      uint8  `json:"qos" yaml:"qos"`
+	Retained bool   `json:"retained" yaml:"retained"`
+	Topic    string `json:"topic" yaml:"topic"`
+	Payload  string `json:"payload" yaml:"payload"`
+}
+
+// EmptyWill returns an empty will, meaning last will message should not be registered.
+func EmptyWill() Will {
+	return Will{}
+}
```

Comment:
Can this be a constant `const EmptyWill = Will{}` instead? I think that's what I've seen the most compared to a function.

Reply:
Only primitive values can be const in Go. If you could have a const struct, it would be the variable that would be const, not its content. Thus, if you had `const EmptyWill = Will{}` and `Will: mqttconf.EmptyWill`, it would presumably still be legal to do `m.conf.will.topic = "asdf"` only this would mutate a struct instance shared between all `MQTTConfig` objects which I think is a potential footgun.